### DISCUSSION TITLE
Shader compiler error with GrayScalePostProcessor example from docs

### DISCUSSION THREAD

**Discussion by mattjennings** [excaliburjs/Excalibur#2357] [Q&A] [answered]
It seems the shader fails to compile the [example grayscale shader from the docs](https://excaliburjs.com/docs/postprocessors#custom-shader-based-post-processors)

Replication: https://codesandbox.io/s/excalibur-shader-i0q8u2?file=/src/index.js

Error:

> Could not link the program: [Fragment shader version does not match other shader versions.]

I know very little about shaders, but the built-in color blindness post processor does work, so perhaps I'm doing something wrong here with this custom one

**Comment by mattjennings** [comment, marked as answer]
I managed to get it working by comparing to the [color blind fragment shader](https://github.com/excaliburjs/Excalibur/blob/b285ede60652ff8273f78a37a1de2bc6a77db485/src/engine/Graphics/PostProcessor/color-blind-fragment.glsl) and updating some of the syntax.

So I guess the example just needs to be updated! (I have no idea if this is the right way to do it - but it works!)

```
#version 300 es
precision mediump float;

// our texture
uniform sampler2D u_image;

// the texCoords passed in from the vertex shader.
in vec2 v_texcoord;

out vec4 fragColor;

void main() {
  vec4 tex = texture(u_image, v_texcoord);
  float avg = 0.2126 * tex.r + 0.7152 * tex.g + 0.0722 * tex.b;
  fragColor = vec4(avg, avg, avg, 1.0);
}
```

**Reply by eonarheim** [reply]
Hi @mattjennings I think this is 100% correct! 

I'll update the docs with that snippet 👍

**Reply by eonarheim** [reply]
https://github.com/excaliburjs/excaliburjs.github.io/commit/ec0b907724405d8edcbeeeec8f878c8da726366c
